src/main/kotlin/net/maizegenetics/phgv2/agc/AnnotateFastas.kt

```diff
@@ -0,0 +1,173 @@
+
+            }
+
+            val workerThreads = (1..threads).map {run ->
+                launch { annotateFasta(inputChannel) }
+            }
+        }
+    }
+
+    // This function takes a list of fasta files and to each idline in each fasta file,
+    // appends " sampleName=<sampleName>" where sampleName is the fasta file name minus the extension
+    // If the idline already contains " sampleName=<sampleName>" then it is not appended.
+    private suspend fun annotateFasta(inputChannel: Channel<InputChannelData>) =
+        withContext (
+            Dispatchers.Default
+        ) {
+            for (entry in inputChannel) {
+                println("annotateFasta: entry = ${entry.sampleName}")
+                val fastaFile = entry.fastaFile
+                val sampleName = entry.sampleName
+                val outputDir = entry.outputDir
+                // Write all lines in the fasta file to the new file.
+                // the sample name is the filename minus the extension
+                // the new file name is the same as the original but it is
+                // written to the outputDir
+                // if  the line starts with >, append " sampleName=${sampleName}" to the line
+
+                // if file was previously compressed, we compress the new file.
+                // Strip off the .gz extension before writing the new non-compressed file.
+                // it wil be added when we re-compress the file.
+                val justName = if (fastaFile.endsWith(".gz")) File(fastaFile).nameWithoutExtension else File(fastaFile).name
+                val newFilename = "${outputDir}/${justName}"
+                File(newFilename).bufferedWriter().use { writer ->
+                    bufferedReader(fastaFile).forEachLine { line ->
+                        if (line.startsWith(">")) {
+                            // Only append if the line does not already contain "sampleName=${sampleName}"
+                            if (!line.contains("sampleName=${sampleName}") )
+                                writer.write("${line} sampleName=${sampleName}\n")
+                            else
+                                writer.write("${line}\n")
+                        } else {
+                            writer.write("${line}\n")
+                        }
+                    }
+                    if (entry.compressed) {
+                        // If the input file was compressed, then the new output file will be compressed
+                        gzipFasta(newFilename)
+                    }
+                }
+            }
+        }
+    fun gzipFasta (file:String) {
+        try {
+
+            // gzip the file
+            // use the -f option to overwrite any existing file
+            myLogger.info("gzipping  file ${file}")
+            val gvcfGzippedFile = file + ".gz"
+            var builder = ProcessBuilder("conda","run","-n","phgv2-conda",
+                "gzip", "-f", file)
```

Comment:
Should we bgzip this instead of gzip?  I don't know if it really matters as they will uncompress the same way, and I am not sure what will happen if we index.